### PR TITLE
Develop   search endpoint

### PR DESCRIPTION
**Pull request by syphzur**

### CODE REVIEW

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ProductController.java [hunk 1]
```diff
@@ -85,7 +86,9 @@ public void addLinks(final ProductModelDto dto)
 
     @GetMapping(path = {"/search/{page}"}, produces = {"application/json"}, params = {"str", "sortedBy"})
     @PreAuthorize("@permissionEvaluatorService.hasPrivilege(authentication, this.authorizationPrefix, 'all' )")
-    public ResponseEntity searchByName(@PathVariable final Integer page, @RequestParam final String str,
+    public ResponseEntity searchByName(@PathVariable final Integer page,
+                                       @RequestParam @Size(min=3, message = "Too short text. Size should be greater than 3.")
```
Comment (dsuchy) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ProductController.java]: than 2

Comment (syphzur) [reply PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ProductController.java]: sonry

Comment (syphzur) [reply PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ProductController.java]: fixed

**Review by margawron — APPROVED** [review]
If it works then you are ready to go.

**Pull request merged**